eater_member_card/models/partner.py

```diff
@@ -1,31 +1,28 @@
 # Copyright 2020 Coop IT Easy SCRL fs
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from odoo import api, fields, models
+from odoo import api, models
 
 
 class Partner(models.Model):
     _inherit = "res.partner"
 
-    parent_barcode = fields.Char(
-        compute="_compute_bar_code", string="Parent Barcode", store=True
-    )
-
     @api.multi
     @api.depends(
         "parent_eater_id",
         "parent_eater_id.barcode",
-        "eater",
+        "eater",  # cf comment hereunder
         "member_card_ids",
     )
     def _compute_bar_code(self):
-        for partner in self:
-            if partner.eater == "eater":
-                partner.parent_barcode = partner.parent_eater_id.barcode
-            elif partner.member_card_ids:
-                for c in partner.member_card_ids:
-                    if c.valid:
-                        partner.barcode = c.barcode
+        parents = self.filtered(lambda p: not p.parent_eater_id)
+        eaters = self.filtered(lambda p: p.parent_eater_id)
+        super(Partner, parents)._compute_bar_code()
+
+        # warning : there is not constraint checking that parent_eater_id
+        # is set if eater field is set to eater
+        for partner in eaters:
+            partner.barcode = partner.parent_eater_id.barcode
```

Comment:
```suggestion
        parents = self.filtered(lambda p: not p.parent_eater_id)
        super(Partner, parents)._compute_bar_code()
        
        eaters = self.filtered(lambda p: p.parent_eater_id)
        # warning : there is not constraint checking that parent_eater_id
        # is set if eater field is set to eater
        for partner in eaters:
            partner.barcode = partner.parent_eater_id.barcode
```
Just reordering lines to make the operations clearer

Reply:
👍